### PR TITLE
Fix description pane losing scroll position

### PR DESCRIPTION
**Pull request by malwilley**
Fixes https://github.com/Microsoft/vscode-pull-request-github/issues/543

Issue was that the webview HTML was being reset every time `update` was called. Now the HTML won't be reset unless the PR has changed.

### CODE REVIEW

**Comment by RMacfarlane** [comment]
Thanks for taking this on!

From my understanding of webviews, when the view becomes hidden and then visible again, the scroll position is not preserved even if the html stays the same. With this change I still see scroll position get reset when switching to a different tab and back to the webview - do you also see this?

I think instead we need to save off this state in the webview - the `pendingCommentText` in `preview-src/index.ts` is a potential example. The markdown extension in VSCode also manages scroll position in its webview

**Comment by malwilley** [comment]
Ah, I see now this was a naive solution! This only fixes the issue listed in the issue (switching between split pane views), not the more general case of switching tabs. I'll work on a more general fix, thanks for the info!

**Comment by malwilley** [comment]
@RMacfarlane I think this should be closer to what we want here. There wasn't an obvious spot for storing the scroll position so I added another private member to `pullRequestOverview.ts` but let me know if that is better kept elsewhere.

File: src/github/pullRequestOverview.ts [hunk 1]
```diff
@@ -224,6 +229,10 @@ export class PullRequestOverviewPanel {
 				return this.checkoutDefaultBranch(message.branch);
 			case 'pr.comment':
 				return this.createComment(message.text);
+			case 'scroll': {
+				this._scrollPosition = message.scrollPosition
```
Comment (RMacfarlane) [inline src/github/pullRequestOverview.ts]: Missing semicolon :)

The linter should run when you do `yarn run watch`, but it's often the top of the output so easy to miss

File: preview-src/index.ts [hunk 2]
```diff
@@ -218,6 +221,16 @@ function addEventListeners(pr: PullRequest): void {
 			branch: pr.repositoryDefaultBranch
 		});
 	});
+
+	window.onscroll = () => {
```
Comment (RMacfarlane) [inline preview-src/index.ts]: This event fires really frequently, so I think we should throttle posting messages.

Maybe something like using `setTimeout` for changes to `pendingCommentText` above

**Comment by RMacfarlane** [comment]
@malwilley This approach looks good! Yeah, I think adding the scroll position to `pullRequestOverview` makes sense. Just had a couple small comments

File: preview-src/index.ts (hunk 2 above)
Comment (malwilley) [reply preview-src/index.ts]: Good thought! I hope it's okay that I've added `debounce` as a dependency for this purpose.

File: package.json [hunk 3]
```diff
@@ -327,8 +328,9 @@
   },
   "dependencies": {
     "@octokit/rest": "^15.9.5",
-    "es6-promise": "^4.2.5",
     "axios": "^0.18.0",
+    "debounce": "^1.2.0",
```
Comment (RMacfarlane) [inline package.json]: Let's update `ThirdPartyNotices.txt` with the license text for this, and then I think this is ready to merge!

**Comment by RMacfarlane** [comment]
Thank you!

**Pull request merged**